libs/s25main/ingameWindows/iwStatistics.cpp

```diff
@@ -278,26 +279,48 @@ void iwStatistics::Draw_()
     DrawStatistic(currentView);
 }
 
-void iwStatistics::DrawPlayerBoxes()
+void iwStatistics::DrawPlayerOverlays()
 {
-    unsigned short startX = 126 - numPlayingPlayers * 17;
-    DrawPoint drawPt = GetDrawPos() + DrawPoint(startX, 68);
+    DrawPoint drawPt = GetDrawPos() + DrawPoint(126 - numPlayingPlayers * 17, 22);
     for(unsigned i = 0; i < gwv.GetWorld().GetNumPlayers(); ++i)
     {
         const GamePlayer& player = gwv.GetWorld().GetPlayer(i);
         if(!player.isUsed())
             continue;
 
+        // Draw the alliances at top of the player image
+        DrawPlayerAlliances(drawPt, player);
+        // Draw player boxes and player status at bottom
         if(activePlayers[i])
+            DrawPlayerBox(drawPt, player);
+        drawPt.x += 34;
+    }
+}
+
+void iwStatistics::DrawPlayerBox(DrawPoint const& drawPt, const GamePlayer& player)
+{
+    auto const playerBoxRect = Rect(DrawPoint(drawPt.x, drawPt.y + 47), Extent(34, 12));
+    auto const playerStatusPosition =
+      DrawPoint(playerBoxRect.getOrigin() + playerBoxRect.getSize() / 2 + DrawPoint(0, 1));
+    DrawRectangle(playerBoxRect, player.color);
+    SmallFont->Draw(playerStatusPosition, getPlayerStatus(player), FontStyle::CENTER | FontStyle::VCENTER,
+                    COLOR_YELLOW);
+}
+
+void iwStatistics::DrawPlayerAlliances(DrawPoint const& drawPt, const GamePlayer& player)
+{
+    constexpr unsigned spacingAllianceRects = 1;
+    constexpr Extent allianceRectExtent(4, 6);
+    constexpr unsigned allianceRectOffset = allianceRectExtent.x + spacingAllianceRects;
```

Comment:
One quick thought: since we still have that issue which is for adding up to 16 player support (which would be a great experience I think), shouldn't we keep that somewhat in mind, which means we might have multiple rows since (currently 1 player can have 7 allies, since 8 players work) there'd be a maximum of 15 allies for each player?

Reply:
For 16 players the statistics window needs a redesign because other things there can at the moment not really handle 8 players right. Therefore i would do this when changing all code to be capable with 16 players and not in this issue.